interaction-manager/src/lib_components/draggable.js

```diff
@@ -0,0 +1,185 @@
+import { clamp } from '../utils/utility-functions';
+import actions from './actions';
+
+const AXIS = ['x', 'y'];
+/**
+ * Makes an element draggable
+ */
+class Draggable {
+    /**
+     *
+     * @param {Object} options
+     * @param {string} options.element
+     * @param {string} options.restrictTo
+     * @param {string} options.dragClass
+     * @param {'x'|'y'} options.lockAxis
+     * @param {function} options.onDragStart
+     * @param {function} options.onDragMove
+     * @param {function} options.onDragEnd
+     */

```

Comment:
We can use typedef - it will also show suggestions
```
/**
 * @typedef {Object} DraggableOptions 
 * @property {string} options.element
 * @property {string} options.restrictTo
 * @property {string} options.dragClass
 * @property {'x'|'y'} options.lockAxis
 * @property {function} options.onDragStart
 * @property {function} options.onDragMove
 * @property {function} options.onDragEnd
 */

 /**
 * @param {DraggableOptions} options
 */
 constructor(options) {}

```

![typedef](https://user-images.githubusercontent.com/19714551/171392168-8de4cbb1-c5d2-48ff-a364-c80b5d397e82.PNG)